packages/cactus-cmd-api-server/src/test/typescript/integration/plugin-import-from-github.test.ts

```diff
@@ -0,0 +1,96 @@
+import path from "path";
+import test, { Test } from "tape-promise/tape";
+import { v4 as uuidv4 } from "uuid";
+import { LogLevelDesc } from "@hyperledger/cactus-common";
+import {
+  PluginImportAction,
+  PluginImportType,
+} from "@hyperledger/cactus-core-api";
+import {
+  ApiServer,
+  AuthorizationProtocol,
+  ConfigService,
+} from "../../../main/typescript/public-api";
+
+const logLevel: LogLevelDesc = "TRACE";
+
+test("can install plugins at runtime with specified version based on imports", async (t: Test) => {
+  const pluginsPath = path.join(
+    __dirname,
+    "../../../../../../", // walk back up to the project root
+    ".tmp/test/test-cmd-api-server/plugin-import-with-npm-install_test/", // the dir path from the root
+    uuidv4(), // then a random directory to ensure proper isolation
+  );
+  const pluginManagerOptionsJson = JSON.stringify({ pluginsPath });
+
+  const configService = new ConfigService();
+
+  const apiServerOptions = configService.newExampleConfig();
+  apiServerOptions.pluginManagerOptionsJson = pluginManagerOptionsJson;
+  apiServerOptions.authorizationProtocol = AuthorizationProtocol.NONE;
+  apiServerOptions.configFile = "";
+  apiServerOptions.apiCorsDomainCsv = "*";
+  apiServerOptions.apiPort = 0;
+  apiServerOptions.cockpitPort = 0;
+  apiServerOptions.grpcPort = 0;
+  apiServerOptions.apiTlsEnabled = false;
+  apiServerOptions.plugins = [
+    {
+      packageName: "@hyperledger/cactus-plugin-keychain-memory",
+      type: PluginImportType.Local,
+      action: PluginImportAction.Install,
+      options: {
+        instanceId: uuidv4(),
+        keychainId: uuidv4(),
+        logLevel,
+        // packageSrc:
+        //   "git+https://github.com/elenaizaguirre/cactus.git#manage-plugins-versions",
+        // packageSrc: "git+https://github.com/hyperledger/cactus.git#28c97d3",
+        // packageSrc:
+        //   "https://gitpkg.now.sh/elenaizaguirre/cactus/packages/cactus-plugin-keychain-memory?main&scripts.postinstall=tsc%20--build%20--verbose",
+        // packageSrc:
+        //   "git+https://github.com/elenaizaguirre/cactus.git?manage-plugins-versions",
+        packageSrc:
+          "https://gitpkg.now.sh/elenaizaguirre/cactus/packages/cactus-plugin-keychain-memory?manage-plugins-versions",
```

Comment:
@petermetz Is there any way to compile a package independently from the rest of the project? This is working because I added the cactus-plugin-keychain-memory dist folder to that branch but I don´t know how to compile only a package

Reply:
@elenaizaguirre My bad, /facepalm 
Since we don't commit the build artifacts to version control, there is no way to get get this done if we adhere to the constraint that I laid out in my earlier message about making the assumption that the sources are built already.

What you could do to test this out anyway is add a dummy package somewhere under the test / resources directory with a dummy plugin implemented in pure JS, with maybe something like this as the directory structure:

```sh
packages/cactus-cmd-api-server/src/test/resources/cactus-plugin-dummy-javascript/index.js
packages/cactus-cmd-api-server/src/test/resources/cactus-plugin-dummy-javascript/package.json
```

Then you could address that via gitpkg just for the sake of being able to verify that the git url plugin loading works as well:

`https://gitpkg.now.sh/elenaizaguirre/cactus/packages/cactus-cmd-api-server/src/test/resources/cactus-plugin-dummy-javascript/?manage-plugins-versions`

Note: for the merged PR it would have to refer to the main branch of the upstream repo not your personal fork so you'll need to break this down into two steps where in the first step you send a PR with the dummy plugin and then in the second step you send a PR where the gitpkg url can be (notice the upstream repo and the `main` branch specifier


`https://gitpkg.now.sh/hyperledger/cactus/packages/cactus-cmd-api-server/src/test/resources/cactus-plugin-dummy-javascript/?main`

Reply:
@petermetz 
Done, I have created #1527 for that purpose, but I needed to add the dist/main folder from a real plugin (instead of a single js file) because when we instantiate a plugin, it should be able to call some plugin methods. Otherwise, server startup will fail